griptape/tasks/toolkit_task.py

```diff
@@ -63,13 +63,25 @@ def memory(self) -> list[BaseToolMemory]:
 
     @property
     def prompt_stack(self) -> PromptStack:
```

Comment:
Implementation totally makes sense but made me think about how we are at some point bound to "prune" during CoT execution but then, once it's done, some memory will get "unpruned." Nothing we can do about it but can potentially affect LLM reasoning.

Reply:
I think this is where some sort of limitless vector based memory could come in handy.